src/plugins/data/public/query/query_string/dataset_service/lib/index_type.ts

```diff
@@ -120,11 +121,10 @@ const fetchDataSources = async (client: SavedObjectsClientContract) => {
     perPage: 10000,
   });
   const dataSources: DataStructure[] = response.savedObjects
-    .filter(
-      (savedObject) =>
-        typeof savedObject.attributes?.dataSourceEngineType === 'string' &&
-        !savedObject.attributes?.dataSourceEngineType?.includes('OpenSearch Serverless')
-    )
+    .filter((savedObject) => {
+      const coercedVersion = semver.coerce(savedObject.attributes.dataSourceVersion);
+      return coercedVersion ? semver.satisfies(coercedVersion, '>=1.0.0') : false;
```

Comment:
ah i see this what we meant with putting version. if this is the case would we be able to go with the [original suggestion](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8758#discussion_r1823351611) of adding an optional property of the dataset type ?